beluga-aws-sdk/src/jobs/mod.rs

```diff
@@ -0,0 +1,306 @@
+use std::collections::HashMap;
+
+use beluga_mqtt::{MqttClient, QoS, Subscriber};
+use chrono::prelude::Utc;
+use futures::prelude::stream::{StreamExt, TryStreamExt};
+
+pub use self::data::JobStatus;
+use self::data::{
+    GetPendingJobsExecutionReq, GetPendingJobsExecutionResp, JobExecution, JobExecutionSummary,
+    StartNextPendingJobExecutionReq, StartNextPendingJobExecutionResp, UpdateJobExecutionReq,
+    UpdateJobExecutionResp,
+};
+use crate::error::Error;
+use crate::Result;
+
+mod data;
+mod datetime;
+
+#[derive(Debug, Clone)]
+pub struct JobsClient {
+    mqtt: MqttClient,
+    subscriber: Subscriber,
+}
+
+impl JobsClient {
+    pub async fn new(mqtt: MqttClient) -> Result<Self> {
+        let subscriber = mqtt
+            .subscribe_many(
+                [
+                    get_accepted(mqtt.thing_name()).as_str(),
+                    get_rejected(mqtt.thing_name()).as_str(),
+                    start_next_accepted(mqtt.thing_name()).as_str(),
+                    start_next_rejected(mqtt.thing_name()).as_str(),
+                ],
+                QoS::AtLeastOnce,
+            )
+            .await?;
+
+        Ok(Self { mqtt, subscriber })
+    }
+
+    pub async fn get(&mut self) -> Result<(Vec<Job>, Vec<Job>)> {
+        self.mqtt
+            .publish(
+                format!("$aws/things/{}/jobs/get", self.mqtt.thing_name()).as_str(),
+                QoS::AtLeastOnce,
+                false,
+                bytes::Bytes::copy_from_slice(&serde_json::to_vec(&GetPendingJobsExecutionReq {
+                    token: format!("get-jobs-{}", self.mqtt.thing_name()),
+                })?),
+            )
+            .await?;
+
+        let packet = self.subscriber.recv().await?;
+
+        if packet.topic == get_accepted(self.mqtt.thing_name()) {
+            let GetPendingJobsExecutionResp {
+                in_progress_jobs,
+                queued_jobs,
+                token,
+                ..
+            } = serde_json::from_slice::<GetPendingJobsExecutionResp>(&packet.payload)?;
+            if token.is_some_and(|token| token == format!("get-jobs-{}", self.mqtt.thing_name())) {
```

Comment:
can we put this in a loop to basically say:

```rust
loop {
  let packet = self.subscriber.recv().await?;
  if !packet_matches_our_request() {
    continue;
  }
  // process job execution resp
}
```

This let's us gracefully deal with the case when there are still unrelated messages in the subscriber.